controller/class.tapestry-controller.php

```diff
@@ -333,32 +339,23 @@ private function _isValidTapestryGroup($groupMetaId)
         return false;
     }
 
-    private function _updateNodes($nodes)
+    private function _addNodes($nodes)
     {
         foreach ($nodes as $node) {
-            if ($this->_isValidTapestryNode($node->id)) {
-                $nodeMetadata = get_metadata_by_mid('post', $node->id)->meta_value;
-                $nodePostId = $nodeMetadata->post_id;
-            } else {
-                $nodePostId = $this->_updatePost($node, 'tapestry_node');
-                $nodeMetadata = $this->_makeMetadata($node, $nodePostId);
-                $node->id = add_post_meta($this->postId, 'tapestry_node', $nodeMetadata);
-            }
+            $nodePostId = $this->_updatePost($node, 'tapestry_node');
+            $metadata = $this->_makeMetadata($node, $nodePostId);
+            $node->id = add_post_meta($this->postId, 'tapestry_node', $metadata);
             update_post_meta($nodePostId, 'tapestry_node_data', $node);
         }
     }
 
-    private function _updateGroups($groups)
+    private function _addGroups($groups)
```

Comment:
It would look cleaner if you change this function to `_addGroup` and accept a single group instead and do the loop over groups in the caller when needed.